### DISCUSSION TITLE
Spot light shadow pass through object

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#456] [Q&A] [answered]
Is this a bug or a unavoidable defect of spot light

The shadow pass through the desk surface and appears under it.

![image](https://user-images.githubusercontent.com/36077492/206869820-df8b7202-b3e6-45e1-adc0-1b52d921706d.png)

**Comment by HSDSZ** [comment]
[shadow pass through.zip](https://github.com/bnpr/Malt/files/10200693/shadow.pass.through.zip)

**Comment by pragma37** [comment, marked as answer]
This is not related to spot lights, really. It's about self-shadows.

Here I applied the same simple material to both objects:
Self-Shadows Enabled:
![imagen](https://user-images.githubusercontent.com/36610999/206870523-13debae3-03d1-415d-9258-132020e2d6c1.png)
Self-Shadows Disabled:
![imagen](https://user-images.githubusercontent.com/36610999/206870537-c2cd5ffa-fbbc-425f-9175-33d725a477d8.png)

It's actually doing exactly what we asked for.
It receives shadows from other objects, but not from itself.

While your desk material doesn't have self-shadows disabled explicitly, it has the *full range* enabled for diffuse, which disables self-shadows.
